doozer/doozerlib/cli/scan_sources.py

```diff
@@ -54,6 +68,142 @@ def run(self):
+            if not rc:
+                # merge succeeded, will push to openshift-priv
+                reconciled = True
+                self.runtime.logger.info('Merged public_upstream/%s into %s', branch_name, metadata.name)
+
+        if not reconciled:
+            # Could not rebase from public upstream: need manual reconciliation. Log a warning and return
+            self.runtime.logger.warning('failed rebasing %s from public upstream: will need manual reconciliation',
+                                        metadata.name)
+            self.issues.append({'name': metadata.distgit_key,
+                                'issue': 'Could not rebase into -priv as it needs manual reconciliation'})
+            return
+
+        if self.dry_run:
+            self.runtime.logger.info('Would have tried reconciliation for %s/%s', repo_name, branch_name)
+            return
+
+        # Try to push to openshift-priv
+        try:
+            exectools.cmd_assert(
+                cmd=['git', 'push', 'origin', branch_name],
+                retries=3)
+
+            # Successfully pushed: mark repo as changed
+            message = f'{metadata.name} reconciled with public openshift/{repo_name}'
+            rebuild_hint = RebuildHint(RebuildHintCode.REBASE_INTO_PRIV, message)
+            self.runtime.logger.info(message)
+
+            if metadata.meta_type == 'rpm':
+                self.add_assessment_reason(metadata, rebuild_hint)
+                self.changing_rpm_metas.add(metadata)
+
+            elif metadata.meta_type == 'image':
+                if rebuild_hint.rebuild:
+                    self.add_image_meta_change(metadata, rebuild_hint)
+
+            else:
+                raise IOError(f'Unsupported meta type: {metadata.meta_type}')
+
+        except ChildProcessError:
+            # Failed pushing to openshift-priv
+            self.runtime.logger.warning('failed pushing to openshift-priv for %s', metadata.name)
+            self.issues.append({'name': metadata.distgit_key,
+                                'issue': 'Failed pushing to openshift-priv'})
+
+    def _do_shas_match(self, public_org, priv_org, repo_name, branch_name) -> bool:
+        """
+        Use GitHub API to check commit SHAs on private and public upstream for a given branch.
+        Return True if they match, False otherwise
+        """
+
+        # Check public commit ID
+        pub_commit = self.github_client.get_repo(f'{public_org}/{repo_name}').get_branch(branch_name).commit
+
+        # Check private commit ID
+        priv_commit = self.github_client.get_repo(f'{priv_org}/{repo_name}').get_branch(branch_name).commit
+
+        if pub_commit.sha == priv_commit.sha:
+            self.runtime.logger.debug('Commit SHAs match on public and priv upstreams for %s', repo_name)
+            return True
+
+        self.runtime.logger.info('Commit SHAs do not match on public and priv upstreams for %s: '
```

Comment:
Latest Commit SHAs? or maybe just "Latest commits do not match.."

Reply:
Updated